File: impls/dotnet/IPCheck.cs
﻿using System;
using System.Net;
using System.Text.Json;

class IPCheck
{
    static void Main(string[] args)
    {
        var input = args[0];
        var addr = IPAddress.Parse(input);

        var data = new {
            to_ipv4 = addr.MapToIPv4().ToString(),
            to_ipv6 = addr.MapToIPv6().ToString(),
            is_unspecified = "<unsupported>",
            is_loopback = IPAddress.IsLoopback(addr),
            is_reserved = "<unsupported>",
        };

        Console.WriteLine("{0}", JsonSerializer.Serialize(data));
    }
}

File: impls/java/IPCheck.java
import java.net.InetAddress;
import javax.json.*;

public class IpCheck {
     public static void main(String []args) throws Exception {
        String input = args[0];
        InetAddress addr = InetAddress.getByName(input);

        JsonObject json = Json.createObjectBuilder()
            .add("to_ipv4", "<unsupported>")
            .add("to_ipv6", "<unsupported>")
            .add("is_unspecified", "<unsupported>")
            .add("is_loopback", addr.isLoopbackAddress())
            .add("is_reserved", "<unsupported>")
            .build();

        System.out.print(json.toString());
     }
}
